2012-09-05  Richard Braun  <rbraun@sceen.net>

       * hurd/hurdsig.c (sigstate_is_global_rcv): Do not return true
       if _HURD_GLOBAL_SIGSTATE is null.

---
 hurd/hurdsig.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/hurd/hurdsig.c b/hurd/hurdsig.c
index 140f951..7840cd5 100644
--- a/hurd/hurdsig.c
+++ b/hurd/hurdsig.c
@@ -159,7 +159,8 @@ _hurd_sigstate_set_global_rcv (struct hurd_sigstate *ss)
 static int
 sigstate_is_global_rcv (const struct hurd_sigstate *ss)
 {
-  return ss->actions[0].sa_handler == SIG_IGN;
+  return (_hurd_global_sigstate != NULL)
+	 && (ss->actions[0].sa_handler == SIG_IGN);
 }
 
 /* Lock/unlock a hurd_sigstate structure.  If the accessors below require
-- 
tg: (2ac13c4..) t/hurdsig-boot-fix (depends on: t/hurdsig-global-dispositions)
